x-pack/plugins/watcher/public/sections/watch_detail/components/watch_detail/watch_history_component.tsx

```diff
@@ -0,0 +1,333 @@
+    initialPageSize: 10,
+    pageSizeOptions: [10, 50, 100],
+  };
+
+  const watchHistoryTimeSpanOptions = [
+    {
+      value: 'now-1h',
+      text: i18n.translate('xpack.watcher.sections.watchHistory.timeSpan.1h', {
+        defaultMessage: 'Last one hour',
+      }),
+    },
+    {
+      value: 'now-24h',
+      text: i18n.translate('xpack.watcher.sections.watchHistory.timeSpan.24h', {
+        defaultMessage: 'Last 24 hours',
+      }),
+    },
+    {
+      value: 'now-7d',
+      text: i18n.translate('xpack.watcher.sections.watchHistory.timeSpan.7d', {
+        defaultMessage: 'Last 7 days',
+      }),
+    },
+    {
+      value: 'now-30d',
+      text: i18n.translate('xpack.watcher.sections.watchHistory.timeSpan.30d', {
+        defaultMessage: 'Last 30 days',
+      }),
+    },
+    {
+      value: 'now-6M',
+      text: i18n.translate('xpack.watcher.sections.watchHistory.timeSpan.6M', {
+        defaultMessage: 'Last 6 months',
+      }),
+    },
+    {
+      value: 'now-1y',
+      text: i18n.translate('xpack.watcher.sections.watchHistory.timeSpan.1y', {
+        defaultMessage: 'Last 1 year',
+      }),
+    },
+  ];
+  const [watchHistoryTimeSpan, setWatchHistoryTimeSpan] = useState<string>(
+    watchHistoryTimeSpanOptions[0].value
+  );
+
+  const columns = [
+    {
+      field: 'startTime',
+      name: i18n.translate('xpack.watcher.sections.watchList.watchTable.startTimeHeader', {
+        defaultMessage: 'Trigger Time',
+      }),
+      sortable: true,
+      truncateText: true,
+      render: (startTime: Moment, item: any) => {
+        const formattedDate = startTime.format();
+        return (
+          <EuiLink
+            className="indTable__link euiTableCellContent"
+            data-test-subj={`watchIdColumn-${formattedDate}`}
+            onClick={() => showDetailFlyout(item)}
+          >
+            {formattedDate}
+          </EuiLink>
+        );
+      },
+    },
+    {
+      field: 'watchStatus.state',
+      name: i18n.translate('xpack.watcher.sections.watchHistory.watchTable.stateHeader', {
+        defaultMessage: 'State',
+      }),
+      sortable: true,
+      truncateText: true,
+      render: (state: string) => {
+        return (
+          <EuiFlexGroup gutterSize="xs" alignItems="center">
+            <EuiFlexItem grow={false}>
+              <WatchActionStatus watchState={state} />
+            </EuiFlexItem>
+            <EuiFlexItem grow={false} className="watchState__message">
+              <EuiText>{state}</EuiText>
+            </EuiFlexItem>
+          </EuiFlexGroup>
+        );
+      },
+    },
+    {
+      field: 'watchStatus.comment',
+      name: i18n.translate('xpack.watcher.sections.watchHistory.watchTable.commentHeader', {
+        defaultMessage: 'Comment',
+      }),
+      sortable: true,
+      truncateText: true,
+      render: (comment: string) => {
+        return <EuiText>{comment}</EuiText>;
+      },
+    },
+  ];
+
+  const onTimespanChange = (e: any) => {
```

Comment:
this should work (and it will be more specific): `(e: React.ChangeEvent<HTMLInputElement>)`